_docs-sources/pipelines/overview/configuration.md

```diff
@@ -1,87 +1,360 @@
+
+At this time, the only supported label that can be set for the first label is `aws_oidc`.
+
+The second label is the name of the authentication profile. This is a user-defined name that can be used to reference the authentication profile in other parts of the configuration. This label must be globally unique.
+
+e.g.
+
+```hcl
+# .gruntwork/authentication_profile.hcl
+authentication_profile "aws_oidc" "profile" {
+  plan_iam_role  = "role-to-assume-for-plans"
+  apply_iam_role = "role-to-assume-for-applies"
+
+  region           = "us-east-1"
+  session_duration = 3600
+}
+
+# some-unit/gruntwork.hcl
+unit {
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = authentication_profile.aws_oidc.profile
+    }
+  }
+}
+```
+
+In this example, the `aws_oidc` authentication profile named `profile` is defined in a file named `authentication_profile.hcl` in the `.gruntwork` directory. The `aws_oidc` authentication block in the `gruntwork.hcl` file in the `some-unit` directory references the `profile` authentication profile.
+
+*Supported Attributes:*
+
+- `plan_iam_role` (Required): The IAM role that Pipelines will assume when running Terragrunt plan commands.
+- `apply_iam_role` (Required): The IAM role that Pipelines will assume when running Terragrunt apply commands.
+- `region` (Optional): The AWS region that Pipelines will use when running Terragrunt commands. If not set, Pipelines will use the default region of `us-east-1`.
+- `session_duration` (Optional): The duration in seconds that the AWS session will be valid for. If not set, Pipelines will use the default session duration of `3600` seconds.
+
+### AWS Accounts Blocks
+
+AWS Accounts blocks are configurations used by `aws-oidc` [authentication](#authentication-blocks) blocks to have commonly re-used AWS account configurations codified and referenced by multiple authentication blocks.
+
+The label applied to an AWS Accounts block is the name of the AWS accounts block. This is a user-defined label for the collection of AWS accounts defined by the block, and must be globally unique.
+
+e.g.
+
+```hcl
+# .gruntwork/aws_accounts.hcl
+aws_accounts "all" {
+  path = "aws/accounts.yml"
+}
+```
+
+In this example, the `all` AWS accounts block is defined in a file named `aws_accounts.yml` within the `.gruntwork` directory. The `all` AWS accounts block references an external file located at `aws/accounts.yml` that contains the definitions of AWS accounts.
+
+DevOps Foundations customers may be familiar with the `accounts.yml` file as a file that is used by Account Factory to define the configurations of AWS accounts. In an effort to reduce the burden of migrating to the new configuration system, Pipelines uses the same schema for the `accounts.yml` file as Account Factory. This means that the `accounts.yml` file that is used by Account Factory can be used by the `aws_accounts` block without modification.
```

Comment:
```suggestion
DevOps Foundations customers may be familiar with the `accounts.yml` file as a file that is used by Account Factory to define the configurations of AWS accounts. This means that the `accounts.yml` file that is used by Account Factory can be used by the `aws_accounts` block without modification.
```
I do not think it is necessary to add that extra info since we also say portability was a factor later on. I would also suggest moving this section below the schema so that it has already been introduced.